packages/angular/src/builders/module-federation-dev-server/module-federation-dev-server.impl.ts

```diff
@@ -1,10 +1,55 @@
 import type { Schema } from './schema';
-import { Workspaces } from '@nrwl/devkit';
+import { ProjectConfiguration, Workspaces } from '@nrwl/devkit';
 import { scheduleTarget } from 'nx/src/adapter/ngcli-adapter';
 import { BuilderContext, createBuilder } from '@angular-devkit/architect';
 import { JsonObject } from '@angular-devkit/core';
 import { join } from 'path';
 import { executeWebpackServerBuilder } from '../webpack-server/webpack-server.impl';
+import { existsSync, readFileSync } from 'fs';
+
+function getDynamicRemotesIfExists(
+  context: BuilderContext,
+  p: ProjectConfiguration
+) {
+  // check for dynamic remotes
+  // we should only check for dynamic based on what we generate
+  // and fallback to empty array
+
+  const standardPathToGeneratedMFManifestJson = join(
+    context.workspaceRoot,
+    p.sourceRoot,
+    'assets/module-federation.manifest.json'
+  );
+  const moduleFederationManifestJson = existsSync(
+    standardPathToGeneratedMFManifestJson
+  )
+    ? readFileSync(standardPathToGeneratedMFManifestJson, 'utf-8')
+    : '{}';
+  if (moduleFederationManifestJson) {
```

Comment:
Returning `'{}'` when the file doesn't exist renders the `if` clause here useless. The `moduleFederationManifestJson` variable will always have a value.

We could just bail out early when the file does not exist.

Reply:
We absolutely could. Good catch